compiler/rustc_middle/src/ty/walk.rs

```diff
@@ -198,6 +198,24 @@ fn push_inner<'tcx>(stack: &mut TypeWalkerStack<'tcx>, parent: GenericArg<'tcx>)
                 | ty::ConstKind::Value(_)
                 | ty::ConstKind::Error(_) => {}
 
+                ty::ConstKind::Expr(expr) => match expr {
+                    ty::Expr::UnOp(_, v) => push_inner(stack, v.into()),
+                    ty::Expr::Binop(_, l, r) => {
+                        push_inner(stack, l.into());
+                        push_inner(stack, r.into())
```

Comment:
we're pushing them on a stack and want to pop the elements in the right order.

For this, earlier children have to be pushed later, so we push them in reverse order. first `r`, then `l`